File: DSA_Level-2/Bit Manipulation/Flip Bits To Convert A To B/FlipBitsToConvertAToB_Solution.js

function flippedBits(a,b){
     
    let flipped_bits=a^b;
    let count=0;

    while(flipped_bits>0){

        let rsb=flipped_bits&-flipped_bits;
        flipped_bits=rsb^flipped_bits;
        ++count;

    }

    console.log(count);

}

flippedBits(57,76);

File: DSA_Level-2/Bit Manipulation/Flip Bits To Convert A To B/FlipBitsToConvertAToB_Solution.java
import java.io.*;
import java.util.*;

class Main {

  public static void main(String[] args){
    Scanner scn = new Scanner(System.in);
    int a = scn.nextInt();
    int b = scn.nextInt();

    int flippedBits=a^b;
    int count=0;
    while(flippedBits!=0){
        int rsb=flippedBits&-flippedBits;
        flippedBits=flippedBits-rsb;
        ++count;
    }
    
    System.out.println(count);
  }

}